From 60cc94d4c3271efcfd787c6276e306ee14cdce72 Mon Sep 17 00:00:00 2001
From: Doug Goldstein <cardoe@cardoe.com>
Date: Tue, 13 Jan 2026 11:08:04 -0600
Subject: [PATCH] hack: for scheduling purposes ignore ports with category
 storage

We want to ignore ports that are in category storage for being eligible
for attachment. This is a hack until TBN arrives and we can utilize
that.

Change-Id: Ibc211ed8e6b9e5933ff657605933efd4c84fb77e
Signed-off-by: Doug Goldstein <cardoe@cardoe.com>
---
 ironic/drivers/modules/network/common.py | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/ironic/drivers/modules/network/common.py b/ironic/drivers/modules/network/common.py
index 04413663c..1ab3e4dbe 100644
--- a/ironic/drivers/modules/network/common.py
+++ b/ironic/drivers/modules/network/common.py
@@ -34,6 +34,21 @@ LOG = log.getLogger(__name__)
 TENANT_VIF_KEY = 'tenant_vif_port_id'
 
 
+def _port_category_hack(port_like_obj):
+    """Hack until we get TBN."""
+
+    category = getattr(port_like_obj, "category", "")
+    if category == "network":
+        # if the category is "network", then we want this port
+        return True
+    elif category == "storage":
+        # if the category is "storage", we don't want this port
+        return False
+    else:
+        # if its unset, backwards compat and attempt to use the port
+        return True
+
+
 def _vif_attached(port_like_obj, vif_id):
     """Check if VIF is already attached to a port or portgroup.
 
@@ -130,6 +145,8 @@ def _get_free_portgroups_and_ports(task, vif_id, physnets, vif_info={}):
             continue
         if not _is_port_physnet_allowed(p, physnets):
             continue
+        if not _port_category_hack(p):
+            continue
         if p.portgroup_id is None and not portgroup_uuid:
             free_port_like_objs.append(p)
         else:
-- 
2.50.1 (Apple Git-155)
